ledger-core/virtual/testutils/utils.go

```diff
@@ -6,9 +6,44 @@
 package testutils
 
 import (
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
 	"github.com/insolar/assured-ledger/ledger-core/vanilla/reflectkit"
 )
 
 func CmpStateFuncs(want, got interface{}) bool {
 	return reflectkit.CodeOf(want) == reflectkit.CodeOf(got)
 }
+
+func WaitSignalsTimed(t *testing.T, timeout time.Duration, signals ...<-chan struct{}) {
+	cleanUp := make(chan struct{})
+	defer close(cleanUp)
+
+	var wg sync.WaitGroup
+	wg.Add(len(signals))
+	for _, c := range signals {
+		go func(signal <-chan struct{}) {
```

Comment:
nope, it's bad behaviour for `server.Journal.WaitAllAsyncCallsDone()`.
It MUST be checked only after `executeDone` is checked (so all async calls were send, but not all were processed)

Reply:
BTW when you call WailAllAsyncCallsDone - it's the point where async call count is >= 0, and if 0 is reached it returns true, so it'll fire
And it failed exact in that place, btw
![image](https://user-images.githubusercontent.com/495636/85199721-6ee08f00-b2fa-11ea-8c2e-2509fa1a0f2f.png)

Reply:
fixed `server.Journal.WaitAllAsyncCallsDone()` by separating calls, didn't realize that aspect at the moment of writing.

Reply:
👍